diff --git a/lib/LaTeXML/Package/babel_support.sty.ltxml b/lib/LaTeXML/Package/babel_support.sty.ltxml
index de32fb6cf..afffecb3d 100644
--- a/lib/LaTeXML/Package/babel_support.sty.ltxml
+++ b/lib/LaTeXML/Package/babel_support.sty.ltxml
@@ -148,6 +148,7 @@ EoTeX
 
 DefPrimitive('\ltx@bbl@select@language{}', sub {
     my ($stomach, $language) = @_;
+    DefMacroI('\cf@encoding', undef, Expand(T_CS('\f@encoding')));
     $language = ToString(Expand($language));
     my $iso = $$bbl_language_map{$language};
     MergeFont(language => $iso) if $iso;
